include_core/omrport.h

```diff
@@ -369,6 +369,13 @@ typedef struct J9PortVmemParams {
 	 *  		- enabled for Linux only,
 	 *  		- If not set, search memory in linear scan method
 	 *  		- If set, scan memory in a quick way, using memory information in file /proc/self/maps. (still use linear search if failed)
+	 * \arg OMRPORT_VMEM_ADDRESS_HINT
+	 *		- enabled for Linux and default pages only,
+	 *		- If not set, search memory page by page until returned address is within desired range
+	 *		- If set, return whatever mmap gives us
+	 *		- this option is based on the observation that mmap would take the preferred address as a hint about where to place the mapping
+	 *		- therefore instead of trying page after page to allocate in the desired region, we stop after the first attempt and return whatever default_pageSize_reserver_memory() gives us as it might give us an address that's good enough
```

Comment:
What if the address is NOT good enough?  There are already STRICT flags so how does this change work if someone has also passed in STRICT?

Reply:
Also what happens for non-default page sizes?  By default we the GC and other components request large pages when they are available.

Reply:
Thanks for taking your time to review this! If the address is not good enough, then one can try again with other options (i.e. with OMRPORT_VMEM_ALLOC_QUICK and larger range in this case). When HINT is set, we don't really care about the STRICT flag anymore. I guess if someone wants STRICT only then they shouldn't be using the HINT option. The whole idea of the hint was based on the observation that mmap would give us a memory address that's close enough even if it's not in the requested range. We don't think the HINT option works for large pages as it uses shmat instead of mmap. The behaviour for large pages stays the same. @mpirvu Hi Marius, correct me if I'm wrong.

Reply:
Harry's response is correct.
If both STRICT and HINT are specified, STRICT is ignored. The difference between HINT and non-STRICT is that with HINT there is only one allocation attempt while in the non-STRICT case there could be many attempts.
Large page allocations are non affected by HINT because the allocation is done with shmat which does not have the notion of a hint like mmap does.

Reply:
Can we make sure that it is documented that HINT has no effect on large page allocations?

Reply:
Okay, updated.